status-im/status-react
# New component Selector > Filter

fixes #14355

### Summary

- [x] Implement component [Selectors > Filter component](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=9961%3A166310&t=FM0i5BsMvIZUECT5-0).
- [x] Use new filter component in the Activity Center. 
- [x] Refactor `selectors.disclaimer` component to follow the new guidelines, also extract it to its own directory with `style.cljs` and `view.cljs` files.
- [x] Fix bug where opening the Activity Center from a fresh start of the app would show the filter toggle disabled (it should be enabled).
- [x] Update guidelines based on PR review. See threads https://github.com/status-im/status-mobile/pull/14650#discussion_r1057822707 and https://github.com/status-im/status-mobile/pull/14650#discussion_r1057824093.
- [x] Add basic quality of life test helpers for writing component tests (the ones using React Native Testing Library).
- [x] Specify how `describe` and `test` macros, as well as global Jest functions `describe` and `test` should be formatted by zprint.

<img src="https://user-images.githubusercontent.com/46027/209694309-b33f1b12-7456-43f4-b809-c5baec888e72.png" width="500" />

### Review notes

**About component test helpers**: this PR is not trying to create a super complete wrapper around Jest and React Native Testing Library (RNTL). I don't even think it's worth it, but while writing component tests, I felt some basic building blocks were missing, so I added the most obvious ones for us. I also decided to use the well known library https://github.com/clj-commons/camel-snake-kebab to convert camel case to keywordized kebab to help interface with RNTL.

**About blurred backgrounds in the preview area**: I did not add the blurred background to the preview screen because it made things look worse due to a known bug where it stays on top of the content. The `@react-native-community/blur` library has hundreds of open issues on Github and it  behaves erratically in Android. One common way to fix the problem I faced is to delay the rendering with animation (which is what other parts of our codebase do), but I didn't bother doing this for the preview area. If you open the Disclaimer preview area in Android you'll see the same problem. Please, take a look at the real usage in the Activity Center to see it working with blurred backgrounds. 

#### Platforms

- Android
- iOS

### Steps to test

Experiment with the quo2.preview screen and double-check the Activity Center filter toggle is correct (see https://github.com/status-im/status-mobile/issues/14355 for details).

status: ready
